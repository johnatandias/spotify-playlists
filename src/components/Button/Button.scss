@import '../../variables';

.button {
  color: $button-text-color;
  background-color: $highlight-color;
  border-width: 0;
  outline: none;
  border-radius: 3px;
  height: auto;
  width: auto;
  padding: 7px;
  font-weight: 500;
}