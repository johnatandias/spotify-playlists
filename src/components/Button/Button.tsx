import React from 'react';
import './Button.scss';

interface Button {
  label: string,
  className?: string,
}

export const Button = ({ label, className = '' }: Button) => (
  <button type="button" className={`button ${className}`}>
    {label}
  </button>
);
