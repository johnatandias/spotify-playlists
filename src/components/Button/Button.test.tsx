import React from "react";
import { shallow } from 'enzyme';
import { Button } from './Button';

describe('Button component', () => {
  it('Renders without crashing', () => {
    const label = 'Click me';
    const wrapper = shallow(<Button label={label} />);
    expect(wrapper.text().includes(label)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders with classname', () => {
    const className = 'my-class';
    const wrapper = shallow(<Button label="Click me" className={className} />);
    expect(wrapper.hasClass(className)).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
