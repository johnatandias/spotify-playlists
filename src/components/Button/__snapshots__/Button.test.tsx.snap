// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Renders with classname 1`] = `ShallowWrapper {}`;

exports[`Button component Renders without crashing 1`] = `ShallowWrapper {}`;
